test/conformance/errorcondition_test.go

```diff
@@ -117,80 +117,97 @@ func TestContainerErrorMsg(t *testing.T) {
 // https://github.com/knative/serving/blob/master/docs/spec/errors.md
 // for the container crashing scenario.
 func TestContainerExitingMsg(t *testing.T) {
-	t.Parallel()
-	clients := setup(t)
-
-	names := test.ResourceNames{
-		Config: test.ObjectNameForTest(t),
-		Image:  "failing",
-	}
-
 	const (
 		// The given image will always exit with an exit code of 5
 		exitCodeReason = "ExitCode5"
 		// ... and will print "Crashed..." before it exits
 		errorLog = "Crashed..."
 	)
 
-	t.Logf("Creating a new Configuration %s", names.Image)
-
-	// This probe is crucial for having a race free conformance test. It will prevent the
-	// pod from becoming ready intermittently.
-	probe := &corev1.Probe{
-		Handler: corev1.Handler{
-			HTTPGet: &corev1.HTTPGetAction{},
+	tests := []struct {
+		Name           string
+		ReadinessProbe *corev1.Probe
+	}{{
+		Name: "http",
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{},
+			},
 		},
-	}
-	if _, err := test.CreateConfiguration(t, clients, names, &test.Options{ReadinessProbe: probe}); err != nil {
-		t.Fatalf("Failed to create configuration %s: %v", names.Config, err)
-	}
-	defer test.TearDown(clients, names)
-	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	}, {
+		Name: "tcp",
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{},
+			},
+		},
+	}}
 
-	t.Log("When the containers keep crashing, the Configuration should have error status.")
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.Name, func(t *testing.T) {
+			t.Parallel()
+			clients := setup(t)
 
-	err := test.WaitForConfigurationState(clients.ServingClient, names.Config, func(r *v1alpha1.Configuration) (bool, error) {
-		cond := r.Status.GetCondition(v1alpha1.ConfigurationConditionReady)
-		if cond != nil && !cond.IsUnknown() {
-			if strings.Contains(cond.Message, errorLog) && cond.IsFalse() {
-				return true, nil
+			names := test.ResourceNames{
+				Config: test.ObjectNameForTest(t),
+				Image:  "failing",
```

Comment:
There are image constants in https://github.com/knative/serving/blob/master/test/conformance/util.go#L42 

I would add this here and reference the constant.

Reply:
Updated to use constants.